### PR TITLE
ToolTip implementation

### PR DESCRIPTION
**Pull request by mishramayank1**
ToolTipBox is a composable that shows a tooltip box with a title and a text. It could also show custom content. It is used to show a tooltip box on a control.

### Validations
Added UI Test Cases

### Screenshots
Bottom Align ToolTip:

image

Top Align ToolTip 
image

Custom content ToolTip:
image

### Pull request checklist

This PR has considered:
- [x] Light and Dark appearances
- [x] Automated Tests
- [x] Documentation and demo app examples
- [x] VoiceOver and Keyboard Accessibility
- [ ] Internationalization and RTL layouts
- [ ] Size classes and window sizes (notched devices, multitasking, different window sizes, etc)

### CODE REVIEW

File: FluentUI.Demo/src/main/java/com/microsoft/fluentuidemo/demos/V2DrawerActivity.kt [hunk 1]
```diff
@@ -383,6 +383,8 @@ private fun getDrawerContent(
         lateinit var context: Context
         AndroidView(
             modifier = Modifier
+
+
```
Comment (praveen970) [inline FluentUI.Demo/src/main/java/com/microsoft/fluentuidemo/demos/V2DrawerActivity.kt:387]: lets rem,ove this file from PR if no change

File: fluentui_core/src/main/java/com/microsoft/fluentui/theme/token/controlTokens/TooltipTokens.kt [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    @Composable
+    open fun backgroundBrush(tooltipInfo: TooltipInfo): Brush =
+        SolidColor(
+            FluentTheme.aliasTokens.neutralBackgroundColor[FluentAliasTokens.NeutralBackgroundColorTokens.BackgroundDarkStatic].value(
+                themeMode = FluentTheme.themeMode
+            )
+        )
+
+    @Composable
+    open fun tipColor(tooltipInfo: TooltipInfo): Color =
```
Comment (praveen970) [inline fluentui_core/src/main/java/com/microsoft/fluentui/theme/token/controlTokens/TooltipTokens.kt:32]: can we make tip a brush just like background brush

File: fluentui_core/src/main/java/com/microsoft/fluentui/theme/token/controlTokens/TooltipTokens.kt [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    }
+
+    @Composable
+    open fun titleColor(tooltipInfo: TooltipInfo): Color =
+        FluentTheme.aliasTokens.neutralForegroundColor[FluentAliasTokens.NeutralForegroundColorTokens.ForegroundLightStatic].value(
+            themeMode = FluentTheme.themeMode
+        )
+
+    @Composable
+    open fun spacingTitleText(tooltipInfo: TooltipInfo): Dp =
```
Comment (praveen970) [inline fluentui_core/src/main/java/com/microsoft/fluentui/theme/token/controlTokens/TooltipTokens.kt:56]: rename to 'titleTextSpacing'

File: fluentui_core/src/main/java/com/microsoft/fluentui/theme/token/controlTokens/TooltipTokens.kt (hunk 2 above)
Comment (mishramayank1) [reply fluentui_core/src/main/java/com/microsoft/fluentui/theme/token/controlTokens/TooltipTokens.kt:32]: Icon & Image does not take brush for tint param

File: fluentui_core/src/main/java/com/microsoft/fluentui/theme/token/controlTokens/TooltipTokens.kt (hunk 3 above)
Comment (mishramayank1) [reply fluentui_core/src/main/java/com/microsoft/fluentui/theme/token/controlTokens/TooltipTokens.kt:56]: Done

File: FluentUI.Demo/src/main/java/com/microsoft/fluentuidemo/demos/V2DrawerActivity.kt (hunk 1 above)
Comment (mishramayank1) [reply FluentUI.Demo/src/main/java/com/microsoft/fluentuidemo/demos/V2DrawerActivity.kt]: removed

**Review by praveen970 — APPROVED** [review]

File: fluentui_icons/src/main/java/com/microsoft/fluentui/icons/__ToolTipIcons.kt
[the remark this answers is not in the document]
Comment (mishramayank1) [reply fluentui_icons/src/main/java/com/microsoft/fluentui/icons/__ToolTipIcons.kt:1]: Fluent system icons have icon which curve at bottom corner, hence cannot be use.

**Comment by andreidiaconu** [comment]
If you want to get rid of the line between the tip and the body, you can consider using a Shape for the background, instead of adding the tip as a vector drawable. This is what we did in our own implementation of Tooltip, and it offers a few other advantages, for example proper shadow or border rendering. 

Screenshot 2023-07-31 103443

Haven't run the demo, my comment is based solely on the render presented in this PR. Feel free to reach out on Teams if this is interesting to you.

**Comment by andreidiaconu** [comment]
Fantastic addition, btw! If this were released just two weeks earlier, we would have been the first to use it.

**Pull request merged**